openverse_catalog/dags/common/loader/loader.py

```diff
@@ -31,3 +31,19 @@ def load_s3_data(
     sql.upsert_records_to_db_table(
         postgres_conn_id, identifier, media_type=media_type, tsv_version=tsv_version
     )
+
+
+def load_from_s3(
+    bucket,
+    key,
+    postgres_conn_id,
+    media_type,
+    tsv_version,
+    identifier,
+):
+    sql.load_s3_data_to_intermediate_table(
+        postgres_conn_id, bucket, key, identifier, media_type
+    )
+    sql.upsert_records_to_db_table(
+        postgres_conn_id, identifier, media_type=media_type, tsv_version=tsv_version
+    )
```

Comment:
I'm not sure the original context/rationale for this decision, but an intermediate table has a few benefits:
1. Isolation from the primary table, so it isn't impacted by the additional time it might take to get the data from S3, only the time it takes to upsert the records from another table
2. We may have loads that affect a large enough number of records that couldn't adequately fit in memory and would be easier to aggregate on disk
3. The intermediate tables get indexed, and those indices can be carried forward into the primary table & reduce the upsertion time

I'm not sure we're at the place _yet_ where we're processing that much data, but we haven't run all the providers regularly so it's possible!

Reply:
Okay, cool! Thanks for the explanation of benefits :grinning: